src/createServer.js

```diff
@@ -1,11 +1,246 @@
 'use strict';
 
-// const express = require('express');
+const express = require('express');
 
 function createServer() {
-  // Use express to create a server
-  // Add a routes to the server
-  // Return the server (express app)
+  const app = express();
+  let users = [];
+  let expenses = [];
+
+  app.post('/users', express.json(), (req, res) => {
+    const { name } = req.body;
+    const newUser = {
+      id: Math.floor(Math.random() * 10),
+      name,
+    };
+
+    if (!name) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    users.push(newUser);
+
+    res.statusCode = 201;
+    res.send(newUser);
+  });
+
+  app.get('/users', express.json(), (req, res) => {
+    if (!users.length) {
+      res.send([]);
+
+      return;
+    }
+    res.statusCode = 200;
+    res.send(users);
+  });
+
+  app.get('/users/:userId', express.json(), (req, res) => {
+    const { userId } = req.params;
+
+    const id = Number(userId);
+    const foundUser = users.find(user => user.id === id);
+
+    if (!foundUser) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    res.statusCode = 200;
+
+    res.send(foundUser);
+  });
+
+  app.patch('/users/:userId', express.json(), (req, res) => {
+    const { userId } = req.params;
+    const foundUser = users.find(user => user.id === +userId);
+
+    if (typeof +userId !== 'number') {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    if (!foundUser) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    const { name } = req.body;
+
+    if (typeof name !== 'string') {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    Object.assign(foundUser, { name });
+
+    res.statusCode = 200;
+
+    res.send(foundUser);
+  });
+
+  app.delete('/users/:userId', express.json(), (req, res) => {
+    const { userId } = req.params;
+    const filterUser = users.filter(user => user.id !== +userId);
+
+    if (filterUser.length === users.length) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    users = filterUser;
+
+    res.sendStatus(204);
+  });
+
+  app.post('/expenses', express.json(), (req, res) => {
+    const { userId } = req.body;
+    const newExpense = {
+      id: Math.floor(Math.random() * 10),
+      ...req.body,
+    };
+
+    if (!users.some(user => user.id === userId)) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    expenses.push(newExpense);
+
+    res.statusCode = 201;
+    res.send(newExpense);
+  });
+
+  app.get('/expenses', express.json(), (req, res) => {
+    const {
+      userId,
+      category,
+      from,
+      to,
+    } = req.query;
+
+    const idNumber = +userId;
```

Comment:
No need to create a separate variable for this purpose, because it is a too simple operation. You could just use `+` or `Number` when it needed.